rust/lance-core/src/io/reader.rs

```diff
@@ -898,13 +898,19 @@ async fn read_struct_array(
     params: &ReadBatchParams,
 ) -> Result<ArrayRef> {
     // TODO: use tokio to make the reads in parallel.
-    let mut sub_arrays: Vec<(FieldRef, ArrayRef)> = vec![];
+    let mut fields: Vec<ArrowField> = vec![];
+    let mut arrays = vec![];
+
     for child in field.children.as_slice() {
         let arr = read_array(reader, child, batch_id, page_table, params).await?;
-        sub_arrays.push((Arc::new(child.into()), arr));
+        fields.push(child.into());
+
+        arrays.push(arr);
     }
 
-    Ok(Arc::new(StructArray::from(sub_arrays)))
+    let sa = StructArray::try_new(fields.into(), arrays, None)?;
+
+    Ok(Arc::new(sa))
```

Comment:
I was getting some panics while debugging that were hard to track down. Since `from` uses `try_new` under the hood, it makes more sense to bubble up the error instead of potentially panicking.